tools/CheckOutputFiles.py

```diff
@@ -155,8 +171,16 @@ def perform_check(self, run_directory):
                         self.check_h5_file(check_h5, self.test_h5_entity,
                                            self.expected_h5_entity)
         self.unit_test.assertTrue(
-            checks_performed, "No checks performed for glob: " +
-            os.path.join(run_directory, self.h5_glob))
+            found_test_entity,
+            "Failed to find the subfile '" + self.test_h5_entity +
+            "' using glob:\n" + os.path.join(run_directory, self.h5_glob) +
+            "\nFiles and entities:\n" + files_and_entities)
+        self.unit_test.assertTrue(
```

Comment:
It looks like python tries to create the error message regardless whether the assert fails or not which is why the unit tests fail.
```
if self.expected_h5_entity is not None:
        self.unit_test.assertTrue(found_expected_entity, ...)
```
fixes it. (It is probably more readable anyway)

Reply:
ah yep :) And agreed, that is easier to read! :) Squashed and rebased